packages/block-editor/src/hooks/block-bindings-panel.scss

```diff
@@ -0,0 +1,12 @@
+.components-panel__block-bindings-panel {
+
+	.components-item {
+		display: flex;
+		justify-content: space-between;
+
+		.components-item__block-bindings-source {
+			color: $gray-700;
+		}
+	}
+
+}
```

Comment:
How are these styles different from the original components? It'd be great to analyze and discuss if this is something specific to this use case or it is something other cases would benefit.

Reply:
@jasmussen Not sure if you'd be able to offer insight here — should we abstract this style so that it's reusable for other features?

To me, this styling here to create a left and right column within the Item group does seem pretty specific to the block bindings panel.

Reply:
I don't know if it is common to other components, but I was thinking of adding properties to handle different use cases.

For example, I can see other components have a `color` property. Does it make sense for this component? Why?

Addtionally, the structure looks really similar to the fonts panel showed [here](https://github.com/WordPress/gutenberg/pull/60706#issuecomment-2107446245). Or even for other things like shadows.

![image](https://github.com/WordPress/gutenberg/assets/34552881/b5681eb2-ac9a-4a01-9673-8276967cfdcc)

Reply:
> Not sure if you'd be able to offer insight here — should we abstract this style so that it's reusable for other features?

Almost certainly, yes. @madhusudhand is working on ItemGroup enhancements as well: https://github.com/WordPress/gutenberg/pull/60706#issuecomment-2110516159

Whether we can get the [ItemGroup](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-itemgroup--docs) component updated in time for this panel to land in 6.6, or you have to use a local temporary setup, that's actually less important — we can componentize and systematize at a later time.

We know this:

* ItemGroup is the right component for this panel, and for shadows, and for fonts.
* We _want_ the ItemGroup component itself to optionally support a "minus" to remove/clear. We need that for fonts, even, for quick removal.
* An optional help-text or "tip" column on the right, feels like a valid prop to have as well. It might be mutually exclusive with some other items, like the reset button. But here's a recent [use case](https://github.com/WordPress/gutenberg/pull/61364#issuecomment-2096146900).

All of those cases may initially be separate things, but conceptually, they're all ItemGroups, and ideally we reach a space where the singular ItemGroup component can support those cases, without local CSS to support it.

Reply:
> For example, I can see other components have a color property. Does it make sense for this component? Why?

@SantosGuillamot Which components are you looking at?

> Addtionally, the structure looks really similar to the fonts panel showed https://github.com/WordPress/gutenberg/pull/60706#issuecomment-2107446245. Or even for other things like shadows.

The structure looks similar but in the case of fonts, they're actually buttons, whereas each item consists of two `span` in this PR at the moment. Once we add the functionality to remove bindings, these items will likely also become buttons instead and perhaps the discrepancy in structure will get resolved then.

<img width="1130" alt="fonts" src="https://github.com/WordPress/gutenberg/assets/5360536/5f5bb679-969a-430c-b13d-e726e5942980">

I'm currently looking at the Shadows structure and see there are some things that may translate to the Bindings Panel — will give that a shot soon.

> Whether we can get the [ItemGroup](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-itemgroup--docs) component updated in time for this panel to land in 6.6, or you have to use a local temporary setup, that's actually less important — we can componentize and systematize at a later time.

Ok, what I'm hearing is to give consolidation a shot with what already exists, but if it doesn't yet, let's just get this merged then consolidate everything later. Let me know if that sounds incorrect!

Reply:
Can we land the proposed UI first and iterate on abstracting CSS later?

Reply:
> @SantosGuillamot Which components are you looking at?

I just clicked something random, but it seems the [Heading component](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-heading--docs) has it. Or we could have variant/styles like in [the button component](https://wordpress.github.io/gutenberg/?path=/docs/components-button--docs).

> An optional help-text or "tip" column on the right, feels like a valid prop to have as well.

I like the idea of having a prop like this.

So, if we include two new props to the component, would we still need the custom CSS? I'm thinking of:
* help-text or "tip".
* color/variant.

Is this something that could be implemented to be reused?

Reply:
> Can we land the proposed UI first and iterate on abstracting CSS later?

This sounds good to me.

> I like the idea of having a prop like this.
> 
> So, if we include two new props to the component, would we still need the custom CSS? I'm thinking of:
> 
> - help-text or "tip".
> - color/variant.
>
> Is this something that could be implemented to be reused?

Seems like they could probably be reused, and seems like this could be good to explore in a separate PR specifically related to the ItemGroup.

> Addtionally, the structure looks really similar to the fonts panel showed https://github.com/WordPress/gutenberg/pull/60706#issuecomment-2107446245. Or even for other things like shadows.

@SantosGuillamot Taking inspiration from shadows, I ended up using the existing HStack component for the Bindings Panel rather than using custom CSS in one of the most recent commits 👍